test/constrained-generics/basic/set2/call-reqfns-from-dfltimpls.chpl

```diff
@@ -0,0 +1,54 @@
+
+interface IFC {
+  proc reqSelf(formal1: Self): Self;
+  proc reqInt(formal2: Self): int;
+  proc dfltSelf(formal3: Self): Self {
+    writeln("in dfltSelf");
+    return reqSelf(reqSelf(formal3));
+  }
+  proc dfltInt(formal4: Self): int {
+    writeln("in dfltInt");
+    return reqInt(dfltSelf(formal4)) * 1000;
+  }
+  proc implicitSelf(formal5: Self) {
+    writeln("in implicitSelf");
+    return dfltSelf(reqSelf(formal5));
```

Comment:
I'm a little bit surprised that we want this to work; I would think we have to specify the return type for these interface functions. however I am OK with it for now.